streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala

```diff
@@ -105,6 +105,9 @@ trait FlinkSubmitTrait extends Logger {
     val retainedOption = CheckpointingOptions.MAX_RETAINED_CHECKPOINTS
     flinkConfig.set(retainedOption, flinkDefaultConfiguration.get(retainedOption))
 
+    //set savepoint.ignore-unclaimed-state parameter
+    flinkConfig.setBoolean(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE, submitRequest.savepointRestoreSettings.allowNonRestoredState())
```

Comment:
As I understand, `SavepointConfigOptions.SAVEPOINT_PATH` and `SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE` can replace the `submitRequest.savepointRestoreSettings`. And the `submitRequest.savepointRestoreSettings` isn't called in application mode, so it's a bug.

So I suggest remove the `submitRequest.savepointRestoreSettings`.

Reply:
Hi @wolfboys , please help double check this solution in your free time, thanks~

Reply:
> Hi @wolfboys , please help double check this solution in your free time, thanks~

👌 I'll check it leater